mlflow/projects/utils.py

```diff
@@ -97,10 +99,32 @@ def _is_file_uri(uri):
     return _FILE_URI_REGEX.match(uri)
 
 
+def _is_git_repo(path):
+    """Returns True if passed-in path is a valid git repository"""
+    import git
+
+    try:
+        git.Repo(path)
+        return True
+    except git.exc.InvalidGitRepositoryError:
+        return False
+
+
 def _is_local_uri(uri):
-    """Returns True if the passed-in URI should be interpreted as a path on the local filesystem."""
-    return not _GIT_URI_REGEX.match(uri)
+    """Returns True if passed-in URI should be interpreted as a path on the local filesystem."""
+    if _GIT_URI_REGEX.match(uri):
+        return False
 
+    parsed_uri = urllib.parse.urlparse(uri).scheme
```

Comment:
```suggestion
    parsed_uri = urllib.parse.urlparse(uri)
```

Sorry, my code suggestion was wrong. We need to remove `.scheme`.

Reply:
@dkapur17 I pushed https://github.com/mlflow/mlflow/pull/6000/commits/e7aab7fba4eebbf3c77b06027c8fd70f6408364e to address this comment :)

Reply:
Awesome!